jerry-core/ecma/builtin-objects/ecma-builtin-helpers-sort.c

```diff
@@ -17,133 +17,137 @@
 #include "ecma-globals.h"
 
 /**
- * Function used to reconstruct the ordered binary tree.
- * Shifts 'index' down in the tree until it is in the correct position.
+ * Function used to merge two arrays for merge sort.
+ * Arrays are stored as below:
+ * First  -> array_a_p [iLeft : iRight - 1]
+ * Second -> array_a_p [iRight : iEnd - 1]
+ * Output -> array_b_p
  *
  * @return ecma value
  *         Returned value must be freed with ecma_free_value.
  */
 static ecma_value_t
-ecma_builtin_helper_array_to_heap (ecma_value_t *array_p, /**< heap data array */
-                                   uint32_t index, /**< current item index */
-                                   uint32_t right, /**< right index is a maximum index */
-                                   ecma_value_t compare_func, /**< compare function */
-                                   const ecma_builtin_helper_sort_compare_fn_t sort_cb) /**< sorting cb */
+ecma_builtin_helper_array_merge_sort_bottom_up (ecma_value_t *array_a_p, /**< arrays to merge */
+                                                uint32_t iLeft, /**< first array begin */
+                                                uint32_t iRight, /**< first array end */
+                                                uint32_t iEnd, /**< second array end */
```

Comment:
JerryScript coding style prefers snake case to camel case, please use left_idx, etc